dsgrid/config/daily_time_dimension_config.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from datetime import timedelta, datetime
+
+import pandas as pd
+
+from dsgrid.dimension.time import DailyTimeRange, TimeBasedDataAdjustmentModel
+from dsgrid.exceptions import DSGInvalidDataset
+from dsgrid.time.types import DailyTimestampType
+from dsgrid.spark.types import (
+    DataFrame,
+    StructType,
+    StructField,
+    IntegerType,
+)
+from dsgrid.utils.timing import timer_stats_collector, track_timing
+from dsgrid.utils.spark import (
+    get_spark_session,
+)
+from .dimensions import DailyTimeDimensionModel
+from .time_dimension_base_config import TimeDimensionBaseConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+class DailyTimeDimensionConfig(TimeDimensionBaseConfig):
+    """Provides an interface to a DailyTimeDimensionModel."""
+
+    @staticmethod
+    def model_class() -> DailyTimeDimensionModel:
+        return DailyTimeDimensionModel
+
+    @track_timing(timer_stats_collector)
+    def check_dataset_time_consistency(self, load_data_df, time_columns) -> None:
+        logger.info("Check DailyTimeDimensionConfig dataset time consistency.")
+        expected_cols = set(self.get_load_data_time_columns())
+        actual_cols = set(time_columns)
+        if expected_cols != actual_cols:
+            msg = (
+                f"DailyTimeDimensionConfig expects columns {expected_cols}, "
+                f"but got {actual_cols}"
+            )
+            raise ValueError(msg)
+
+        # Get the year column name (time_year, weather_year, or model_year)
+        year_col = self.get_year_column()
+        month_col = "time_month"
+        day_col = "time_day"
```

Comment:
@lixiliu, does representative time allow user discretion for all column names, or does it require certain column names?

Reply:
I see that it requires specific column names: (`month`, `day_of_week`, `hour`) OR (`month`, `hour`, `is_weekday`). Per that patter, I'll change my column names to `month`, `day`.